FactoryWeather/App/VIP Settings/SettingsRouter.swift

```diff
@@ -0,0 +1,35 @@
+//
+//  SettingsRouter.swift
+//  FactoryWeather
+//
+//  Created by Matej Korman on 09/01/2019.
+//  Copyright © 2019 Matej Korman. All rights reserved.
+//
+
+import Foundation
+
+protocol SettingsRoutingLogic {
+    func getNewWeather(selectedLocation: Location?)
```

Comment:
I would name this method a bit more detailed. When I first saw this one, I was rather confused as to why does the `router` perform a `business use-case` of fetching the weather - but, in fact, it does not. 

Usually when I have methods on the `router` that notify the `router delegate` of something, I use the `requestSomething` method naming. In example, `getNewWeather` would become something alongside: `requestNewWeatherData` or `requestWeatherDataReload`, and such. 

Much more concise and understandable.